docs/guides/aws-private-link-workspace.md

```diff
@@ -0,0 +1,261 @@
+- `storage_configuration_id` - Databricks workspace storage configuration ID
+
+This guide is provided as-is and you can use this guide as the basis for your custom Terraform module.
+
+To get started with AWS PrivateLink integration, this guide takes you throw the following high-level steps:
+- Initialize the required providers
+- Configure AWS objects
+  - A subnet dedicated to your VPC relay and workspace endpoints
+  - A security group dedicated to your VPC endpoints
+  - Two AWS VPC endpoints
+- Workspace Creation
+
+## Provider initialization
+
+Initialize [provider with `mws` alias](https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations) to set up account-level resources. See [provider authentication](../index.md#authenticating-with-hostname,-username,-and-password) for more details.
+
+```hcl
+terraform {
+ required_providers {
+   databricks = {
+     source  = "databrickslabs/databricks"
+     version = "0.4.9"
+   }
+   aws = {
+     source = "hashicorp/aws"
+     version = "3.49.0"
+   }
+ }
+}
+
+provider "aws" {
+ region = var.region
+}
+
+// initialize provider in "MWS" mode for provisioning workspace with AWS PrivateLink
+provider "databricks" {
+ alias    = "mws"
+ host     = "https://accounts.cloud.databricks.com"
+ username = var.databricks_account_username
+ password = var.databricks_account_password
+}
+```
+
+Define the required variables
+
+```hcl
+variable databricks_account_id {}
+variable databricks_account_username {}
+variable databricks_account_password {}
+variable vpc_id {}
+variable network_name {}
+variable region {}
+variable existing_network_sg {}
+variable existing_network_subnets {}
+
+variable workspace_vpce_service {}
+variable relay_vpce_service {}
+
+variable vpc_cidr_block {}
+variable vpce_cidr {}
+
+// Use the Databricks Account API 2.0 to retrieve these two IDs below - https://docs.databricks.com/dev-tools/api/latest/account.html
+variable credentials_id {}
+variable storage_configuration_id {}
+
+locals {
+ prefix = "private-link-ws"
+}
+
+locals {
+workspace_subnet_1 = (
+ element(split(",", var.existing_network_subnets), 1)
+)
+workspace_subnet_2 = (
+ element(split(",", var.existing_network_subnets), 2)
+)
+}
+```
+
+## Configure AWS objects
+The first step is to create the required AWS objects:
+- A subnet dedicated to your VPC endpoints
+- A security group dedicated to your VPC endpoints and satisfying required inbound/outbound TCP/HTTPS traffic rules on ports 443 and 6666, respectively.
+- Lastly, creation of the private access settings and workspace.
+
+```hcl
+resource "aws_subnet" "vpce" {
+  vpc_id     = var.vpc_id
+  cidr_block = var.vpce_cidr
+
+  tags = {
+    Name = "vpce subnet for workspace"
+  }
+}
+
+resource "aws_route_table" "my_vpc_private" {
+    vpc_id = var.vpc_id
+
+    tags = {
+        Name = "Local Route Table for Isolated Private Subnet"
+    }
+}
+
+resource "aws_route_table_association" "my_vpc_us_east_1a_private" {
```

Comment:
`my_vpc_us_east_1a_private` this is too specific name and it leaks name of the region, which is a variable. fix it.